Lekce8HW/Vehicle.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Drawing;
+
+public abstract class Vehicle
+{
+	protected float Mileage { get; set; }
+	protected int ID { get; set; }
+
+	protected bool EngineStatus { get; set; }
+	protected ConsoleColor VehicleColor { get; set; }
+	protected ConsoleColor[] _collorPallete = { ConsoleColor.Blue, ConsoleColor.Green, ConsoleColor.Yellow, ConsoleColor.Magenta, ConsoleColor.Red };
+	public ConsoleColor ColorPallete(int i)
+	{
+		return _collorPallete[i];
+	}
+
+
+
+    public Vehicle()
+	{
+        EngineStatus = false;
+        Mileage = 0;
+		VehicleColor = ConsoleColor.White;
+    }
+
+	public abstract void Repaint(ConsoleColor color);
+	public virtual void EngineOnOff()	//Asi určitě by bylo lepší udělat 2 rozdílné funkce a už vůbec ne dávat zhasnutí motoru do ELSE místo ELSE IF.
+    {									//Navíc jsem si chtěl vyzkoušet pracovat s obrácením bool hodnot pomocí !. A obecně řešit problémy, které s tímto přístupem nastanou.
+		if (EngineStatus == false)
+		{
+            Console.WriteLine("Engine is STARTING...");
+			EngineStatus = !EngineStatus;
+		}
+		else 
+		{
+            Console.WriteLine("Engine SHUT DOWN");
+            EngineStatus = !EngineStatus;
+		}
```

Comment:
Uhledneji napsano by to vypadalo asi takto:
```suggestion
		if (EngineStatus)
		{
			Console.WriteLine("Engine SHUT DOWN");
		}
		else 
		{
			Console.WriteLine("Engine is STARTING...");
		}
		EngineStatus = !EngineStatus;
```

Reply:
u bool hodnot se nemusi porovnavat `== false` nebo `== true` apod. Je dobre primo pracovat s tou hodnotou jako takovou

A to "flipnuti"
`EngineStatus = !EngineStatus;`
se ma vykonat ve vsech pripadech (jak v if, tak else branchi), takze je dobre ho z tech bloku vyndat. Kdykoliv predchazis duplicitam, tak zlepsujes kod a sanci na chybu :)